Runtime/Scripts/MobiledgeXSettings.cs

```diff
@@ -1,14 +1,167 @@
+    public class ReadOnlyAttribute : PropertyAttribute{  }
+
+    [CustomPropertyDrawer(typeof(ReadOnlyAttribute))]
+    public class ReadOnlyDrawer : PropertyDrawer
+    {
+        public override float GetPropertyHeight(SerializedProperty property,
+                                                GUIContent label)
+        {
+            return EditorGUI.GetPropertyHeight(property, label, true);
+        }
+
+        public override void OnGUI(Rect position,
+                                   SerializedProperty property,
+                                   GUIContent label)
+        {
+            GUI.enabled = false;
+            EditorGUI.PropertyField(position, property, label, true);
+            GUI.enabled = true;
+        }
+    }
+#endif
+    // MatchingEngine AppPort is not Serializable
+    // Protocol enum is used to override LProto easier to read
+    [System.Serializable]
+    public class Port
+    {
+        /// <summary>
+        /// Tag to label certain tags, you can retrieve ports by tag 
+        /// </summary>
+        public string tag;
+
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public Protocol protocol;
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public string fqdn_prefix;
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public string path_prefix;
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public int internal_port;
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public int public_port;
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public int end_port;
+#if UNITY_EDITOR
+        [ReadOnly]
+#endif
+        public bool TLS;
+
+        public Port(AppPort port)
+        {
+            switch (port.proto)
+            {
+                case LProto.L_PROTO_HTTP:
+                case LProto.L_PROTO_TCP:
+                    this.protocol = Protocol.TCP;
+                    break;
+
+                case LProto.L_PROTO_UDP:
+                    this.protocol = Protocol.UDP;
+                    break;
+
+                case LProto.L_PROTO_UNKNOWN:
+                default:
+                    this.protocol = Protocol.Unknown;
+                    break;
+
+            }
+
+            this.fqdn_prefix = port.fqdn_prefix;
+            this.path_prefix = port.path_prefix;
+            this.internal_port = port.internal_port;
+            this.public_port = port.public_port;
+            this.end_port = port.end_port;
+            this.TLS = port.tls;
+            this.tag = TLS==true?"Secure ":"" + protocol + " "+port.public_port;
+        }
+        public override string ToString()
+        {
+            return TLS == true ?"Secure":"" + protocol + public_port;
+        }
+    }
+    public enum Protocol
+    {
+        TCP,
+        UDP,
+        Unknown
+    }
+
+     public class MobiledgeXSettings: ScriptableObject
+    {
+        public string orgName;
+        public string appName;
+        public string appVers;
+        public string authPublicKey;
```

Comment:
is there any way to add an "optional" note next to this? Also, developer may also want uniqueID and uniqueIDType

Reply:
Regarding Unique ID and UniqueIDType posted a question for that on the Slack channel,
Regarding making (AuthPublicKey) optional its not included in the Editor Window (used for Setup) its exposed MobiledgeX Settings(Similar to extra config options on Photon Settings)

Reply:
Those are for MEL only, so...no developer should want uniqueID and uniqueIDType, including us outside the SDK runtime.

Reply:
Ok uniqueID discussion continuing in slack

Reply:
authpublic key should also be in the Set-Up. If a user supplies an authkey when creating their app (in console or mcctl), they must supply a JWT token with orgname, appname, appvers information in that token otherwise RegisterClient will fail and user will not be able to setup